core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -1217,7 +1217,8 @@ public static IndexMetaData fromXContent(XContentParser parser) throws IOExcepti
      * {@link #isIndexUsingShadowReplicas(org.elasticsearch.common.settings.Settings)}.
      */
     public static boolean isOnSharedFilesystem(Settings settings) {
-        return settings.getAsBoolean(SETTING_SHARED_FILESYSTEM, isIndexUsingShadowReplicas(settings));
+        Version version = settings.getAsVersion(SETTING_VERSION_CREATED, Version.CURRENT);
+        return settings.getAsBooleanLenientForPreEs6Indices(version, SETTING_SHARED_FILESYSTEM, isIndexUsingShadowReplicas(settings));
```

Comment:
I use the `indexCreatedVersion` of `IndexMetaData` now (just had to turn this into an instance method) and have also provided an overloaded version without parameters.